tests/phpunit/tests/functions/wpCacheSetLastChanged.php

```diff
@@ -0,0 +1,38 @@
+<?php
+/**
+ * Tests for the wp_cache_set_last_changed function.
+ *
+ * @group functions.php
+ *
+ * @covers ::wp_cache_set_last_changed
+ */#
+class Tests_functions_wpCacheSetLastChanged extends WP_UnitTestCase {
+
+	/**
+	 * check the cache_set_last_changed is set.
+	 *
+	 * @ticket 59737
+	 */
+	public function test_wp_cache_set_last_changed() {
+		$group = 'group_name';
+
+		$this->assertSame( wp_cache_set_last_changed( $group ), wp_cache_get( 'last_changed', $group ) );
+	}
+
+	/**
+	 * Check the action is called.
+	 *
+	 * @ticket 59737
+	 */
+	public function test_wp_cache_set_last_changed_action_is_called() {
+
+		add_action(
+			'wp_cache_set_last_changed',
+			function ( $group ) {
+				$this->assertSame( 'group_name', $group );
+			}
+		);
```

Comment:
It's best to use `MockAction` for this, in this form if the action doesn't fire the test won't fail but instead throw a warning that it does not contain any tests.

This test contains some prior art you can work from (ignore the calls removing the actions, they're not needed)

https://github.com/WordPress/wordpress-develop/blob/f0d53f8100d5bc274b126995e5f08a509a70d91c/tests/phpunit/tests/post.php#L395-L421

Reply:
Thanks updated